digdag-standards/src/main/java/io/digdag/standards/command/KubernetesCommandExecutor.java

```diff
@@ -276,7 +293,7 @@ CommandStatus getCommandStatusFromKubernetes(final CommandContext context,
             final InputStream in = outConfigStorage.getContentInputStream(outputArchiveKey);
             ProjectArchives.extractTarArchive(context.getLocalProjectPath(), in); // runtime exception
         }
-        else if (defaultPodTTL.isPresent() && isRunningLongerThanTTL(previousStatusJson)) {
+        else if (isRunningLongerThanOutConfigStorageExpiration(previousStatusJson) || (defaultPodTTL.isPresent() && isRunningLongerThanTTL(previousStatusJson))) {
```

Comment:
I guess you are using a non-blocking operator such as `py>`. In this case, IIRC, `(defaultPodTTL.isPresent() && isRunningLongerThanTTL(previousStatusJson))` will never become true. In addtion, this block will not execute when user/[WorkflowExecutionTimeoutEnforcer](https://github.com/treasure-data/digdag/blob/953036db19567ac24b1fc38e9fb802dd866c952e/digdag-server/src/main/java/io/digdag/server/WorkflowExecutionTimeoutEnforcer.java#L179-L180) cancel-requested to the attempt. Is it OK/acceptable for you?  
We also had the same kind of issue. Thus we introduce the `cleanup` function to the Operator and CommnadExecutor interfaces and implements them in PyOperatorFactory and EcsCommandExecutor. Please see #1480, and this function helps to perform the cleanup process if the attempt received cancel-requested. I feel it is a good idea to handle it by using the same way to avoid such an issue. 
How about adding it to `KubernetesCommandExecutor` as well? How do you think about that?